src/components/SearchAndFilter/SearchAndFilter.tsx

```diff
@@ -35,36 +36,86 @@ const SearchAndFilter: React.FC<filterProps> = ({data, columns, filterTypes}) =>
         }
       })
     }
-    setIsLoading(false)
-  }, [data, filterTypes, isLoading])
-  
-  
-  const filterData = useCallback((matchingItems:object[]) => {
-    let result:object[] = matchingItems
-    if (filterTypes) {
+  }, [allData, filterTypes])
+
+  const createFilterOptions = (filterType: string) => {
+    const availableOptions = determineAvailableFilterValues(filterType)
+
+    const options = availableOptions.map((option, i) => {
+        if (availableOptions.indexOf(option) === i) {
+          return <option value={option.toString()}>{option}</option>
+        }
+      })
+
+    return (
+      <>
+      <label htmlFor={`${filterType}-selector`}>{filterType}</label>
+      <select 
+        id={`${filterType}-selector`} 
+        onChange={(event) => {
+          dispatch({type: filterType, value: event.target.value})
+          messageIfSearchFailed()
+        }}
+        value={selectedFilters[filterType]}
+      >
+        <option value="">All</option>
+        {options}
+      </select>
+      </>
+    )
+  }
+
+  const filterData = useCallback(() => {
+    let result:Array<{id:string}> = []
+    
+    const someFilterIsSelected = ():boolean => {
+      return Object.values(selectedFilters).some(filter => filter !== 'All')
+    }
+    
+    if (filterTypes && someFilterIsSelected()) {
       filterTypes.forEach(filter => {
-        if (selectedFilters[filter] !== 'All' && filter !== 'search') {
-          result = result.filter((item:stringKeyOptions) => item[filter].includes(selectedFilters[filter]))
+        if (selectedFilters[filter] !== 'All') {
+          const dataToSearch = result.length > 0 ? result : allData
```

Comment:
This line serves to prevent one filter from overwriting another filter's results when searching in tandem. It doesn't feel very readable and I'd like to come back to it.